lib/spack/spack/test/compilers.py

```diff
@@ -251,3 +260,82 @@ def test_xl_r_flags():
 def test_clang_version_detection(version_str, expected_version):
     version = spack.compilers.clang.Clang.detect_version_from_str(version_str)
```

Comment:
Why is this still `detect_version_from_str` when all the others are `extract_version_from_output`? Can we make them more uniform?

Reply:
You're right. It makes sense to use the same name and override the function.